Update collections.md (#134)
LaravelDaily/laravel-tips: collections.md

- [Multiple Collection Methods in a Row](#multiple-collection-methods-in-a-row)
- [Calculate Sum with Pagination](#calculate-sum-with-pagination)

> - [Serial no in foreach loop with pagination](#serial-no-in-foreach-loop-with-pagination)

Reviewer: This deletion is not mentioned in your commit or PR. It is similar but different from the tip that follows it and should probably be kept. 

```suggestion
- [Serial no in foreach loop with pagination](#serial-no-in-foreach-loop-with-pagination)
- [Higher order collection methods](#higher-order-collection-methods)
```

- [Higher order collection message](#higher-order-collection-message)
- [Get an existing key or insert a value if it doesn't exist and return the value](#get-an-existing-key-or-insert-a-value-if-it-doesnt-exist-and-return-the-value)
- [Static times method](#static-times-method)

> - [Paginate from array method](#Paginate-from-array-method)

Reviewer: Update name to match revised tip

```suggestion
- [Paginated Collections](#paginated-collections)
```

it will solve the issue of next pages(?page=2&...) index count from continue.

> 

Reviewer: This deletion is not mentioned in your commit or PR. It is similar but different from the tip that follows it and should probably be kept.

```suggestion

### Higher order collection methods

Collections have higher order methods, this are methods that can be chained , like `groupBy()` , `map()` ... Giving you a fluid syntax. This example calculates the
price per group of products on an offer.

```php
$offer = [
        'name'  => 'offer1',
        'lines' => [
            ['group' => 1, 'price' => 10],
            ['group' => 1, 'price' => 20],
            ['group' => 2, 'price' => 30],
            ['group' => 2, 'price' => 40],
            ['group' => 3, 'price' => 50],
            ['group' => 3, 'price' => 60]
        ]
];

$totalPerGroup = collect($offer['lines'])->groupBy('group')->map(fn($group) => $group->sum('price'));
```
```

// Output: [01-04-2022, 02-04-2022, ..., 07-04-2022]
```

> ### Paginate from array method
> 
> This method enables you to do all things that the official paginate do but with Arrays.
> 
> ```php
> use Illuminate\Pagination\LengthAwarePaginator;
> ```
> 
> ```php
> public function paginate($perPage, $total = null, $page = null, $pageName = 'page')
>     {
>         $page = $page ?: LengthAwarePaginator::resolveCurrentPage($pageName);
> 
>         return new LengthAwarePaginator(
>             $this->forPage($page, $perPage),
>             $total ?: $this->count(),
>             $perPage,
>             $page,
>             [
>                 'path' => LengthAwarePaginator::resolveCurrentPath(),
>                 'pageName' => $pageName,
>             ]
>         );
>     }
> ```
> 
> Tip given by [@Teacoders](https://twitter.com/Teacoders/status/1509447909602906116)

Reviewer: This tip is currently incomplete and does not contain enough information to inform someone how to paginate an array. I've seen this exact code elsewhere and it contains a more complete example of the class implementation.  In its current form it cannot be used to directly paginate an array.  Additionally, there probably aren't a bunch of people using custom Collections, so the macro method is probably more helpful. In either case, it would have been appropriate to link your source.

Your tip also pushes the previous tips attribution line out of place.

```suggestion
Tip given by [@Teacoders](https://twitter.com/Teacoders/status/1509447909602906116)

### Paginated Collections

Add this to you AppServiceProvider.php files boot() method, or similar provider, and you can then paginate any collection or array with ease. 

```php
        use Illuminate\Pagination\LengthAwarePaginator;

        /**
         * Paginate a standard Laravel Collection.
         *
         * @param  int  $perPage
         * @param  int  $total
         * @param  int  $page
         * @param  string  $pageName
         * @return LengthAwarePaginator
         * @source https://gist.github.com/simonhamp/549e8821946e2c40a617c85d2cf5af5e
         */
        Collection::macro('paginate', function ($perPage, $total = null, $page = null, $pageName = 'page') {
            $page = $page ?: LengthAwarePaginator::resolveCurrentPage($pageName);

            /** @var Collection $this */
            return new LengthAwarePaginator(
                $this->forPage($page, $perPage),
                $total ?: $this->count(),
                $perPage,
                $page,
                [
                    'path' => LengthAwarePaginator::resolveCurrentPath(),
                    'pageName' => $pageName,
                ]
            );
        });
```

Example Usage: 
`collect([ ... ])->paginate( 20 );`

```
